planner/core/property_cols_prune.go

```diff
@@ -32,10 +32,11 @@ func (ds *DataSource) preparePossibleProperties() [][]*expression.Column {
 		if len(path.idxCols) == 0 {
 			continue
 		}
-
-		result = append(result, path.idxCols)
+		result = append(result, make([]*expression.Column, len(path.idxCols)))
```

Comment:
Project will change the slice returned from child. So it will change the content of `path.IdxCols`. So we need a shallow copy here.
Most changes in tests are related with these bug codes